ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityRewardedAd.java

```diff
@@ -0,0 +1,272 @@
+            Log.w(UnityAdapter.TAG, ids + " cannot be empty.");
+
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void onUnityAdsReady(String placementId) {
+        // Unity Ads is ready to show ads for the given placementId.
+    }
+
+    @Override
+    public void onUnityAdsStart(String placementId) {
+        // Unity Ads video ad started playing. Send Video Started event if this is a rewarded
+        // video.
+        if (mMediationRewardedAdCallback != null) {
+            mMediationRewardedAdCallback.onVideoStart();
+        }
+    }
+
+    @Override
+    public void onUnityAdsClick(String s) {
+        // Unity Ads ad clicked.
+        if (mMediationRewardedAdCallback != null) {
+            mMediationRewardedAdCallback.reportAdClicked();
+        }
+    }
+
+    @Override
+    public void onUnityAdsPlacementStateChanged(String placementId,
+                                                UnityAds.PlacementState oldState,
+                                                UnityAds.PlacementState newState) {
+        // This callback is not forwarded to Google Mobile Ads SDK. onUnityAdsError should be used
+        // to forward Unity Ads SDK state to Google Mobile Ads SDK.
+    }
+
+    @Override
+    public void onUnityAdsFinish(String placementId, UnityAds.FinishState finishState) {
+        // Unity Ads ad closed.
+        if (mMediationRewardedAdCallback != null) {
+            // Reward is provided only if the ad is watched completely.
+            if (finishState == UnityAds.FinishState.COMPLETED) {
+                mMediationRewardedAdCallback.onVideoComplete();
+                // Unity Ads doesn't provide a reward value. The publisher is expected to
+                // override the reward in AdMob console.
+                mMediationRewardedAdCallback.onUserEarnedReward(new UnityReward());
+                mMediationRewardedAdCallback.onAdClosed();
+            }
+            else if (finishState == UnityAds.FinishState.ERROR)
+            {
+                String message = "Unity Ads finished in an error state" +
+                        " while trying to show rewarded ad for placement ID '" + placementId + "'.";
+                Log.w(UnityAdapter.TAG, message);
+                mMediationRewardedAdCallback.onAdFailedToShow(message);
+            }
+            else {
+                mMediationRewardedAdCallback.onAdClosed();
+            }
+        }
+    }
+
+    @Override
+    public void onUnityAdsError(UnityAds.UnityAdsError unityAdsError, String errorMessage) {
+        // Unity Ads ad failed to show.
+        String logMessage = "Failed to show Rewarded ad for placement ID '" + getPlacementId() +
+                "' from Unity Ads. Error: " + unityAdsError.toString() + " - " + errorMessage;
+        Log.w(UnityAdapter.TAG, logMessage);
+        if (mMediationRewardedAdCallback != null) {
+            mMediationRewardedAdCallback.onAdFailedToShow(logMessage);
+        }
```

Comment:
`mMediationRewardedAdCallback.onAdFailedToShow` will be called twice, if error occurs.

Reply:
I'll take it out of onUnityAdsFinish